src/ancient_clj/io/s3.clj

```diff
@@ -1,25 +1,38 @@
 (ns ancient-clj.io.s3
-  (:require [ancient-clj.io.xml :as xml]
-            [aws.sdk.s3 :as s3])
-  (:import [com.amazonaws.services.s3.model AmazonS3Exception]))
+  (:require [ancient-clj.io.xml :as xml])
+  (:import (com.amazonaws.auth AWSCredentialsProvider BasicAWSCredentials DefaultAWSCredentialsProviderChain)
+           (com.amazonaws.services.s3 AmazonS3ClientBuilder)
+           (com.amazonaws.services.s3.model AmazonS3Exception)))
 
 (def ^:private valid-content-types
   #{"text/xml" "application/xml"})
 
+(defn ^:private s3-get-object!
+  [client bucket key]
```

Comment:
Ah, I just realized that then we'd be creating a new client for every artifact and thatÄs no good either.

Maybe we can wrap the client in a `delay` and dereference it within `s3-get-object!`?